dj/service_clients.py

```diff
@@ -143,24 +142,33 @@ def materialize(  # pylint: disable=too-many-arguments
             GenericMaterializationInput,
             DruidMaterializationInput,
         ],
-    ) -> MaterializationOutput:
+    ) -> MaterializationInfo:
         """
         Post a request to the query service asking it to set up a scheduled materialization
         for the node. The query service is expected to manage all reruns of this job. Note
         that this functionality may be moved to the materialization service at a later point.
         """
-        response = self.requests_session.post(  # pragma: no cover
+        response = self.requests_session.post(
             "/materialization/",
             json=materialization_input.dict(),
         )
-        result = response.json()  # pragma: no cover
-        return MaterializationOutput(**result)  # pragma: no cover
+        if not response.ok:  # pragma: no cover
+            return MaterializationInfo(urls=[])
+        result = response.json()
+        return MaterializationInfo(**result)
 
-    def get_materializations(self, node_name, materialization_name):
+    def get_materialization_info(
+        self,
+        node_name,
+        materialization_name,
+    ) -> MaterializationInfo:
         """
-        Gets a list of materializations for the node and materialization config name.
+        Gets materialization info for the node and materialization config name.
         """
         response = self.requests_session.get(
             f"/materialization/{node_name}/{materialization_name}/",
+            timeout=3,
         )
-        return response.json()
+        if not response.ok:
+            return MaterializationInfo(output_tables=[], urls=[])
```

Comment:
What does it mean when the other service reports multiple output tables? Is the idea that it will include all previous/expired materializations, for example previous days for a materialization that's recreated daily? Then one of those tables would be set as the `AvailabilityState` for a given node which would then be used during the SQL build/compile, is that all right?

Reply:
So the multiple output tables is actually to handle the Druid use case, where we would have both the output of the Spark query and the Druid datasource after it's ingested there. Arguably we don't need the former and can just have a single output, but I think it can be nice to have the list as a "we produced these artifacts" type of record. But yes, only one will get set as the availability state.

Reply:
I see, that makes sense although I think it could be confusing when someone sets a druid materialization and sees both the name of a druid datasource and a spark table. Would there even be a way to tell that one is an intermediate spark table? As I'm writing this I'm starting to think the spark table is just an implementation detail of how the data gets to to druid which can vary wildly across different query services.

Maybe we should stick to maintaining the assumption that an output table for a given materialization config is always a table for the given engine+catalog combinations, the engine being selected in the materialization config and the catalog being the one for the node. That sounds like a guarantee that's up to the query service to provide so I don't think I'm suggesting any changes to this PR 😄